utils/cloudinfo/cloudinfo.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudinfo
+
+import (
+	"net/http"
+)
+
+type CloudProvider int
+
+const (
+	GCE CloudProvider = iota
+	AWS
+	BAREMETAL
+	OTHER
+)
+
+func GetCloudProvider() CloudProvider {
```

Comment:
Consider creating an interface for this method. It will be useful for unit testing.